### PR TITLE
doc: 17.2.7 Release Notes

### PR DESCRIPTION
**Pull request by yuriw**
## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/releases/quincy.rst [hunk 1]
```diff
@@ -5,6 +5,593 @@ Quincy
...
+v17.2.7 Quincy
+==============
+
+This is the seventh backport release in the Quincy series. We recommend
+that all users update to this release.
+
+Notable Changes
+---------------
+
+* `ceph mgr dump` command now displays the name of the mgr module that
```
Comment (anthonyeleven) [inline doc/releases/quincy.rst:17]: s/mgr/Manager/

File: doc/releases/quincy.rst [hunk 2]
```diff
@@ -5,6 +5,593 @@ Quincy
...
+  client was shown in the `active_clients` array.
+
+* mClock Scheduler: The mClock scheduler (default scheduler in Quincy) has
+  undergone significant usability and design improvements to address the slow
+  backfill issue. Some important changes are:
+  * The 'balanced' profile is set as the default mClock profile because it
+    represents a compromise between prioritizing client IO or recovery IO. Users
+    can then choose either the 'high_client_ops' profile to prioritize client IO
+    or the 'high_recovery_ops' profile to prioritize recovery IO.
+  * QoS parameters like reservation and limit are now specified in terms of a
```
Comment (anthonyeleven) [inline doc/releases/quincy.rst:29]: s/like/including/

**Review by zdover23 — APPROVED** [review]

File: doc/releases/quincy.rst [hunk 3]
```diff
@@ -5,6 +5,593 @@ Quincy
...
+    osd_mclock_cost_per_byte_usec_*) have been removed. The cost of an operation
+    is now determined using the random IOPS and maximum sequential bandwidth
+    capability of the OSD's underlying device.
+  * Degraded object recovery is given higher priority when compared to misplaced
+    object recovery because degraded objects present a data safety issue not
+    present with objects that are merely misplaced. Therefore, backfilling
+    operations with the 'balanced' and 'high_client_ops' mClock profiles may
+    progress slower than what was seen with the 'WeightedPriorityQueue' (WPQ)
+    scheduler.
+  * The QoS allocations in all the mClock profiles are optimized based on the above
```
Comment (anthonyeleven) [inline doc/releases/quincy.rst:41]: s/all the/all//

File: doc/releases/quincy.rst [hunk 4]
```diff
@@ -5,6 +5,593 @@ Quincy
...
+    fixes and enhancements.
+  * For more detailed information see:
+    https://docs.ceph.com/en/quincy/rados/configuration/mclock-config-ref/
+
+* RGW: S3 multipart uploads using Server-Side Encryption now replicate correctly in
+  multi-site. Previously, the replicas of such objects were corrupted on decryption.
+  A new tool, ``radosgw-admin bucket resync encrypted multipart``, can be used to
+  identify these original multipart uploads. The ``LastModified`` timestamp of any
+  identified object is incremented by 1ns to cause peer zones to replicate it again.
```
Comment (anthonyeleven) [inline doc/releases/quincy.rst:50]: s/1ns/one nanosecond/

File: doc/releases/quincy.rst [hunk 5]
```diff
@@ -5,6 +5,593 @@ Quincy
...
+  For multi-site deployments that make any use of Server-Side Encryption, we
+  recommended running this command against every bucket in every zone after all
+  zones have upgraded.
+
+* CEPHFS: MDS evicts clients which are not advancing their request tids which causes
```
Comment (anthonyeleven) [inline doc/releases/quincy.rst:55]: CephFS

File: doc/releases/quincy.rst (hunk 1 above)
Comment (zdover23) [reply doc/releases/quincy.rst:17]: Incorporated.

File: doc/releases/quincy.rst (hunk 2 above)
Comment (zdover23) [reply doc/releases/quincy.rst:29]: Incorporated.

File: doc/releases/quincy.rst (hunk 4 above)
Comment (zdover23) [reply doc/releases/quincy.rst:50]: Incorporated.

File: doc/releases/quincy.rst (hunk 5 above)
Comment (zdover23) [reply doc/releases/quincy.rst:55]: Incorporated.

**Comment by zdover23** [comment]
I take back all the things I said were incorporated:

>  To github.com:yuriw/ceph.git
>  ! [remote rejected]         wip-yuriw-release-17.2.7-main -> wip-yuriw-release-17.2.7-main (permission denied)
> error: failed to push some refs to 'github.com:yuriw/ceph.git'

File: doc/releases/quincy.rst (hunk 5 above)
Comment (yuriw) [reply doc/releases/quincy.rst:55]: @anthonyeleven @zdover23 I don't feel we need to edit release notes as they were entered by the authors.
What's our rule for this?

Comment (anthonyeleven) [reply doc/releases/quincy.rst:55]: I don't know if there's a rule, but IMHO content changes and spelling/grammar are different.

File: doc/releases/quincy.rst [hunk 6]
```diff
@@ -5,6 +5,593 @@ Quincy
...
+* `ceph mgr dump` command now displays the name of the mgr module that
+  registered a RADOS client in the `name` field added to elements of the
+  `active_clients` array. Previously, only the address of a module's RADOS
+  client was shown in the `active_clients` array.
+
+* mClock Scheduler: The mClock scheduler (default scheduler in Quincy) has
+  undergone significant usability and design improvements to address the slow
+  backfill issue. Some important changes are:
+  * The 'balanced' profile is set as the default mClock profile because it
+    represents a compromise between prioritizing client IO or recovery IO. Users
```
Comment (ljflores) [inline doc/releases/quincy.rst:26]: Sphinx says there is an unexpected indentation on this line.

File: doc/releases/quincy.rst (hunk 2 above)
Comment (ljflores) [inline doc/releases/quincy.rst:29]: Sphinx also says there is an unexpected indent here.

File: doc/releases/quincy.rst (hunk 6 above)
Comment (ljflores) [reply doc/releases/quincy.rst:26]: (line 26)

File: doc/releases/quincy.rst (hunk 2 above)
Comment (ljflores) [reply doc/releases/quincy.rst:29]: (line 29)

File: doc/releases/quincy.rst (hunk 6 above)
Comment (zdover23) [reply doc/releases/quincy.rst:26]: This is caused because there isn't a blank line before line 25. I know that that might sound crazy, but that's Sphinx.

Comment (ljflores) [reply doc/releases/quincy.rst:26]: Noted.. fixing

**Comment by ljflores** [comment]
> I take back all the things I said were incorporated:
> 
> > To github.com:yuriw/ceph.git
> > ! [remote rejected]         wip-yuriw-release-17.2.7-main -> wip-yuriw-release-17.2.7-main (permission denied)
> > error: failed to push some refs to 'github.com:yuriw/ceph.git'

It's on the ceph fork, not Yuri's. Try just pushing to Ceph! (origin)

**Comment by ljflores** [comment]
@zdover23 ^

File: doc/releases/quincy.rst (hunk 6 above)
Comment (yuriw) [reply doc/releases/quincy.rst:26]: What exactly has to be changed?

File: doc/releases/quincy.rst (hunk 2 above)
Comment (yuriw) [reply doc/releases/quincy.rst:29]: What exactly has to be changed?

**Comment by ljflores** [comment]
@zdover23 @yuriw I added another commit and squashed them to fix one of the formatting errors. JFYI in case you add further commits (rebase with the new changes first).

**Comment by zdover23** [comment]
@ljflores: 
[zdover@fedora releases]$ git push origin wip-yuriw-release-17.2.7-main
Enumerating objects: 18, done.
Counting objects: 100% (18/18), done.
Delta compression using up to 8 threads
Compressing objects: 100% (7/7), done.
Writing objects: 100% (12/12), 37.63 KiB | 7.53 MiB/s, done.
Total 12 (delta 10), reused 6 (delta 5), pack-reused 0

**Comment by zdover23** [comment]
@ljflores: I also tried to push to "ceph" (ceph/ceph), but I don't have the permissions to do that.

File: doc/releases/quincy.rst (hunk 5 above)
Comment (zdover23) [reply doc/releases/quincy.rst:61]: @yuriw: We'll take care of it. You don't have to do anything. I promise that we won't change the technical content of anything we touch.

File: doc/releases/quincy.rst (hunk 6 above)
Comment (zdover23) [reply doc/releases/quincy.rst:27]: @yuriw: I'll take care of it. Don't worry about anything.

**Comment by zdover23** [comment]
@ljflores @yuriw: I am satisfied now that this is ready for publication. Shall we merge and backport it?

File: doc/releases/quincy.rst [hunk 7]
```diff
@@ -5,6 +5,602 @@ Quincy
...
+* rgwlc: prevent lc for one bucket from exceeding time budget (`pr#53561 <https://github.com/ceph/ceph/pull/53561>`_, Matt Benjamin)
+* test/cli-integration/rbd: iSCSI REST API responses aren't pretty-printed anymore (`pr#52283 <https://github.com/ceph/ceph/pull/52283>`_, Ilya Dryomov)
+* test: correct osd pool default size (`pr#51804 <https://github.com/ceph/ceph/pull/51804>`_, Nitzan Mordechai)
+* test: monitor thrasher wait until quorum (`pr#51801 <https://github.com/ceph/ceph/pull/51801>`_, Nitzan Mordechai)
+* tools/ceph-dencoder: Fix incorrect type define for trash_watcher (`pr#51779 <https://github.com/ceph/ceph/pull/51779>`_, Chen Yuanrun)
+* tools/cephfs-data-scan: support for multi-datapool (`pr#50522 <https://github.com/ceph/ceph/pull/50522>`_, Mykola Golub)
+* tools/cephfs: add basic detection/cleanup tool for dentry first damage (`pr#52245 <https://github.com/ceph/ceph/pull/52245>`_, Patrick Donnelly)
+* tools/cephfs: include lost+found in scan_links (`pr#50783 <https://github.com/ceph/ceph/pull/50783>`_, Patrick Donnelly)
+* v17.2.6 (`pr#50895 <https://github.com/ceph/ceph/pull/50895>`_, Ceph Release Team, Casey Bodley, Radosław Zarzyński)
```
Comment (idryomov) [inline doc/releases/quincy.rst:597]: This was pointed out in the past -- including a tag of the `N-1` release in the release notes for the `N` release is confusing.  If it's a problem with the release notes script itself (as opposed to how it's invoked), please make a note to delete this line manually for now.

File: doc/releases/quincy.rst [hunk 8]
```diff
@@ -5,6 +5,602 @@ Quincy
...
+* doc: Update mClock QOS documentation to discard osd_mclock_cost_per\_\* (`pr#54080 <https://github.com/ceph/ceph/pull/54080>`_, tanchangzhi)
+* doc: update multisite doc (`pr#51401 <https://github.com/ceph/ceph/pull/51401>`_, parth-gr)
+* doc: update rados.cc (`pr#52968 <https://github.com/ceph/ceph/pull/52968>`_, Zac Dover)
+* doc: update README.md (`pr#52642 <https://github.com/ceph/ceph/pull/52642>`_, Zac Dover)
+* doc: update README.md install procedure (`pr#52680 <https://github.com/ceph/ceph/pull/52680>`_, Zac Dover)
+* doc: update test cluster commands in README.md (`pr#53350 <https://github.com/ceph/ceph/pull/53350>`_, Zac Dover)
+* doc: Use `ceph osd crush tree` command to display weight set weights (`pr#51350 <https://github.com/ceph/ceph/pull/51350>`_, James Lakin)
+* docs: fix nfs cluster create syntax (`pr#52424 <https://github.com/ceph/ceph/pull/52424>`_, Paul Cuzner)
+* docs: Update the Prometheus endpoint info (`pr#51287 <https://github.com/ceph/ceph/pull/51287>`_, Paul Cuzner)
+* exporter: user only counter dump/schema commands for extacting counters (`pr#50749 <https://github.com/ceph/ceph/pull/50749>`_, Avan Thakkar)
```
[the remark this answers is not in the document]
Comment (cbodley) [reply doc/releases/quincy.rst:348]: i didn't see a PR for this, i've prepared the revert in https://github.com/ceph/ceph/pull/54169

[the remark this answers is not in the document]
Comment (yuriw) [reply doc/releases/quincy.rst:349]: @idryomov @cbodley I've updated this PR with the latest reverts
Pls double check

[the remark this answers is not in the document]
Comment (yuriw) [reply doc/releases/quincy.rst:349]: @idryomov is it a new issue you are saying blocking 17.2.7 or the one we resolved?

[the remark this answers is not in the document]
Comment (idryomov) [reply doc/releases/quincy.rst:349]: The revert is merged, so we are good!

**Review by idryomov — APPROVED** [review]

**Comment by ljflores** [comment]
> @ljflores @yuriw: I am satisfied now that this is ready for publication. Shall we merge and backport it?

@zdover23 let me verify the content

**Comment by ljflores** [comment]
https://ceph--54153.org.readthedocs.build/en/54153/releases/quincy/

**Comment by ljflores** [comment]
Several revert PRs need to be removed.. doing that now

**Comment by ljflores** [comment]
I removed the "Revert" PRs and their initial counterparts, and changed the release date to Monday.

**Review by ljflores — APPROVED** [review]

**Comment by ljflores** [comment]
Docs are ready to go in my eyes; we are waiting on the final "build containers" step, which is in progress today.

**Pull request merged**